tools/local-sensor/main.go

```diff
@@ -124,10 +228,11 @@ func main() {
 
 	spyCentral.ConnectionStarted.Wait()
 
-	<-time.Tick(*durationFlag)
+	log.Printf("Running scenario for %f minutes\n", localConfig.Duration.Minutes())
+	<-time.Tick(localConfig.Duration)
```

Comment:
If you decide to implement the suggestion above, you could handle the error signal here other than in a separate `goroutine` like it's being done now.

```golang
select {
   case <-time.Tick(localConfig.Duration):
      // ran scenario for long enough, continue
   case <- errSig.Done():
      // something went wrong 
}
```

Reply:
Makes sense. I will do this in the new task.